src/citrine/jobs/waiting.py

```diff
@@ -126,14 +127,19 @@ def wait_while_validating(
                                         interval=interval)
 
 
+# Might need to add GenerativeDesignExecution here.
```

Comment:
You could perhaps derive from a common base class like ExecutionCollection? Then you don't need an explicit Union type.

Reply:
I wouldn't worry about it just for the purposes of type annotation.

Reply:
Yeah agree, can just plop the new type it into the Union annotation there